index.js

```diff
@@ -890,6 +916,66 @@ class Batch {
   }
 }
 
+class EntryWatcher extends ReadyResource {
+  constructor (bee, key) {
+    super()
+
+    this.index = bee._entryWatchers.push(this) - 1
+    this.bee = bee
+
+    this.key = key
+    this.node = null
+
+    this._forceUpdate = false
+    this._debouncedUpdate = debounce(this._processUpdate.bind(this))
+  }
+
+  async _open () {
+    this.node = await this.bee.get(this.key)
+  }
+
+  _close () {
+    const top = this.bee._entryWatchers.pop()
+    if (top !== this) {
+      top.index = this.index
+      this.bee._entryWatchers[top.index] = top
+    }
+  }
+
+  async _onappend () {
+    await this._debouncedUpdate()
+  }
+
+  async _processUpdate () {
+    if (!this.opened) await this.ready()
+
+    const force = this._forceUpdate
+    this._forceUpdate = false
+
+    let newNode
+    try {
+      newNode = await this.bee.get(this.key)
+    } catch (e) {
+      if (e.code === 'SNAPSHOT_NOT_AVAILABLE') {
+        // There was a truncate event before the get resolved
+        // So this handler will run again anyway
+        return
+      }
+      this.emit('error', e)
+    }
+
+    if (force || newNode?.seq !== this.node?.seq) {
+      this.node = newNode
+      this.emit('update')
+    }
+  }
+
+  async _ontruncate () {
+    this._forceUpdate = true
+    await this._debouncedUpdate(true)
```

Comment:
remove true here

Reply:
you shouldn't await here also and ontruncate shouldnt be async, as there is no place for the async context to bubble up to.

ie its a background promise, so the debounced function is NOT allowed to throw outside a panic (which it also never does atm)

Reply:
same for onappend